manifests/user.pp

```diff
@@ -32,28 +29,25 @@
 ) {
   include ::st2::params
 
-  $_robots_group_name = $st2::params::robots_group_name
   $_packs_group_name = $st2::params::packs_group_name
-
-  ensure_resource('group', $_robots_group_name, {
-    'ensure' => present,
-  })
+  $_ssh_dir = "/home/${name}/.ssh"
 
   ensure_resource('group', $_packs_group_name, {
     'ensure' => present,
   })
 
   if $create_sudo_entry {
-    ensure_resource('sudo::conf', $_robots_group_name, {
+    ensure_resource('sudo::conf', $name, {
       'priority' => '10',
-      'content'  => "%${_robots_group_name} ALL=(ALL) NOPASSWD: SETENV: ALL",
+      # note: passes in $name variable into template
+      'content'  => template('st2/etc/sudoers.d/user.erb'),
     })
```

Comment:
So before running the `Puppet-st2`, I have the following:
```
ubuntu@ubuntu16:~$ sudo ls -la /etc/sudoers.d/
total 16
drwxr-x---  2 root root 4096 Aug  8 15:35 .
drwxr-xr-x 91 root root 4096 Aug  8 15:35 ..
-r--r-----  1 root root  123 Aug  8 15:35 90-cloud-init-users
-r--r-----  1 root root  958 Mar 30  2016 README
```

Running the Puppet provisioner removes that file, - that's why it breaks exsiting sudoers.